daemon/containerd/image_push.go

```diff
@@ -261,7 +261,7 @@ func (i *ImageService) getPushDescriptor(ctx context.Context, img containerdimag
 
 	switch len(presentMatchingManifests) {
 	case 0:
-		return ocispec.Descriptor{}, errdefs.NotFound(fmt.Errorf("no suitable image manifest found for platform %s", *platform))
+		return ocispec.Descriptor{}, errdefs.NotFound(fmt.Errorf("no suitable image manifest found for platform %s", platforms.FormatAll(*platform)))
```

Comment:
We may be able to improve these errors further, as they're still somewhat "technical". Perhaps improve them to more clearly describe "the image was found, but not for the given platform variant"

I'll keep that for a future change though (will open a ticket for improving)

Reply:
I went looking for it as I was sure I'd seen something like this, and realized we do have https://github.com/docker/moby/blob/810c7c1dce5bbf76af4ed5c6bac8c47fead4f6c6/daemon/containerd/image.go#L27 in the engine. Maybe we should think about exporting it.